### PR TITLE
AAD authority validation threading and serialization test

### PR DESCRIPTION
**Pull request by RPangrle**
This test is designed to make sure that if two acquireToken calls are made simultaneously with no cache that only one of them hits the network for the validation/metadata check

### CODE REVIEW

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m [hunk 1]
```diff
@@ -242,4 +243,99 @@ - (void)testValidateAuthority_whenHostUnreachable_shouldFail
...
+    
+    // Network Setup
+    NSArray *metadata = @[ @{ @"preferred_network" : @"login.contoso.com",
+                              @"preferred_cache" : @"login.contoso.com",
+                              @"aliases" : @[ @"sts.contoso.com", @"login.contoso.com"] } ];
+    dispatch_semaphore_t validationSem = dispatch_semaphore_create(0);
+    ADTestURLResponse *validationResponse = [ADTestAuthorityValidationResponse validAuthority:authority withMetadata:metadata];
+    [validationResponse setWaitSemaphore:validationSem];
+    ADTestURLResponse *tokenResponse1 = [self adResponseRefreshToken:TEST_REFRESH_TOKEN
+                                                           authority:@"https://login.contoso.com/common"
```
Comment (oldalton) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: nit: why not use authority variable here instead?

Comment (RPangrle) [reply ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: 👍

**Comment by RPangrle** [comment]
It appears our network mock isn't entirely thread safe, and this test is hitting that on macOS, at least occasionally... I might have to fix that in a separate PR.

**Comment by RPangrle** [comment]
I've fixed the threading issues in #1047

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m [hunk 2]
```diff
@@ -203,15 +205,14 @@ - (void)testAcquireToken_whenAuthorityInvalid_shouldReturnError
...
-    NSString* authority = @"https://login.windows.cn/MSOpenTechBV.onmicrosoft.com";
-
+    NSString* authority = @"https://login.windows.net/contoso.com";
     
     ADAuthorityValidation* authorityValidation = [[ADAuthorityValidation alloc] init];
     ADRequestParameters* requestParams = [ADRequestParameters new];
     requestParams.authority = authority;
     requestParams.correlationId = [NSUUID UUID];
     
-    NSURL* requestURL = [ADAuthorityValidationRequest urlForAuthorityValidation:authority trustedHost:s_kTrustedAuthority];
```
Comment (antrix1989) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: Should we delete s_kTrustedAuthority? Seems it's not used anywhere.

Comment (RPangrle) [reply ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: 👍

**Pull request merged**